app/models/restart_signal_handler.rb

```diff
@@ -1,11 +1,18 @@
 # JobQueue locks a mutex, hence the need for a separate SignalHandler thread
 # Self-pipe is also best practice, since signal handlers can themselves be interrupted
-class SignalHandler
+class RestartSignalHandler
+  class << self
+    alias listen new
```

Comment:
splits responsibility/knowledge into 2 places that have nothing to do with each other